src/compiler/utilities.ts

```diff
@@ -4179,6 +4179,14 @@ namespace ts {
         return node.kind === SyntaxKind.Identifier || isPropertyAccessEntityNameExpression(node);
     }
 
+    export function getLeftMostIdentifierOfEntityName(node: EntityName) {

```

Comment:
I was very surprised that this didn’t already exist. I searched for EntityName and QualifiedName all over this file and didn’t find anything. 🤔

Reply:
`getFirstIdentifier` from checker ?

Reply:
JS handling in the binder uses something similar, but it may be getRightmostIdentifierOfPropertyAccessExpression.